src/sentry/auth/access.py

```diff
@@ -375,14 +331,18 @@ def from_user(user, organization=None, scopes=None, is_superuser=False):
     return from_member(om, scopes=scopes, is_superuser=is_superuser)
 
 
-def from_member(member, scopes=None, is_superuser=False):
+def from_member(
+    member: OrganizationMember,
+    scopes: Optional[Iterable[str]] = None,
+    is_superuser: bool = False,
+) -> Access:
     # TODO(dcramer): we want to optimize this access pattern as its several
     # network hops and needed in a lot of places
     requires_sso, sso_is_valid = _sso_params(member)
 
-    team_list = member.get_teams()
+    team_list = frozenset(member.get_teams())
```

Comment:
@RyanSkonnord what's the advantage of using a frozen set here? Concerned this might have downstream reprocussions.

Reply:
Primarily, simplicity/consistency with the type declaration (`teams: FrozenSet[Team]`). If the flexibility is needed, we could change that declaration to `Sequence`. But I'd also consider it advantageous (less bug-prone) to have the `Access` values be immutable once created.